server/src/main/java/org/apache/kafka/server/share/CachedSharePartition.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.server.share;
+
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.message.ShareFetchResponseData;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.common.requests.ShareFetchRequest;
+import org.apache.kafka.common.requests.ShareFetchResponse;
+import org.apache.kafka.common.utils.ImplicitLinkedHashCollection;
+
+/**
+ * A cached share partition.
+ * <p>
+ * The broker maintains a set of these objects for each share session.
+ * When a share fetch request is made, any partitions which are not explicitly
+ * enumerated in the fetch request are loaded from the cache. Similarly, when a
+ * share fetch response is being prepared, any partitions that have not changed and
+ * do not have errors are left out of the response.
+ * <p>
```

Comment:
I would remove this paragraph. I does beg the question of why we store a Uuid object (which I think we should).

Reply:
Done.